frontend/test/cypress/integration/product-list/parts_results_view.spec.ts

```diff
@@ -0,0 +1,63 @@
+describe('parts page results view', () => {
+   const user = cy;
+
+   beforeEach(() => {
+      user.visit('/Parts');
+   });
+
+   it('all products have price visible', () => {
+      user
+         .findByTestId('list-view-products')
+         .children('article')
+         .each((productListItem) => {
+            user
+               .wrap(productListItem)
+               .findByTestId('product-price')
+               .should('be.visible');
+         });
+   });
+
+   /**
+    * In ProductListItem, we set noOfLines prop to 3
+    * to truncate the product description text. This
+    * tests each product description to be less than 4
+    * lines.
+    */
+   it('product description is truncated', () => {
+      // Counting text lines https://stackoverflow.com/questions/783899
+      user.get('[data-testid="product-short-desc"]').each((descriptionDiv) => {
```

Comment:
How come we are using `get` instead of `findByTestId` in this line?

Reply:
It turns out `findByTestId` fails if there are multiple elements with `data-testid="..."` on the page. `findByTestId` seems to expect to find only one element. Therefore, I had to use `get` which works with getting multiple elements
![fail](https://user-images.githubusercontent.com/58952979/178333638-138173a3-3522-4f30-89df-4cc5e30733cc.PNG)

Reply:
@batbattur I suggest not to rely too much on test ids.. I use test ids as the last resort generally.. I try to test based on accessible elements. This way tests also validates accessibility. 
While writing tests I've tried to follow [these principles](https://testing-library.com/docs/guiding-principles), especially point 2. The closest way to select an input like a user would do is to search using accessible names

Reply:
> It turns out findByTestId fails if there are multiple elements with data-testid="..." on the page. findByTestId seems to expect to find only one element. Therefore, I had to use get which works with getting multiple elements

I was wrong here. Just found that there is `findAllByTestId` in [cypress-testing-library.](https://github.com/testing-library/cypress-testing-library/blob/main/types/index.d.ts#L155)

<hr>

> @batbattur I suggest not to rely too much on test ids.. I use test ids as the last resort generally.. I try to test based on accessible elements. This way tests also validates accessibility.

@dhmacs Sounds good, and thank you for the advice! From this [article](https://betterprogramming.pub/why-you-should-avoid-testing-react-components-with-test-ids-ee50d20d37d2), I found that `getByRole`, `getByLabelText`, or `getByText` are recommended instead of using data-testid. 

In our case, if I want to get the product short description text element, how would you recommend? Maybe should I add `aria-label="product_short_desc"` instead of adding `data-testid="product_short_desc"` to the text?
![short_description](https://user-images.githubusercontent.com/58952979/178832371-ca524db7-6379-43cb-bf9a-2f6e01b6fc33.jpg)

Reply:
@batbattur I'd start by analyzing what the test is trying to do. From what I understand here we're trying to test if the product description is truncated. Imo I wouldn't test this behaviour at all, as is not really critical for the functioning of the e-commerce.
Having said that, `aria-label` is something that is pronounced by the screen reader, so "product_short_desc" will not sound nice. In this case you can add a screen only span with "product description" content and and `id` and reference this `id` in the `aria-labelledby` attribute of the description paragraph. This way the description will have an accessible name. But I think using a test id is also a valid option here, as we probably don't need to give accessible names to every little thing.

Reply:
Dropped the test in https://github.com/iFixit/react-commerce/pull/422/commits/bc0835f7b30fcd28c34ed48fde336ce1e2936f01.